### PR TITLE
Worksheet D (incomplete version)

### PR DESCRIPTION
**Pull request by Mindengine76**
I was not able to complete the brief  and so have submitted an incomplete program. I am more than happy to re attempted this assignment over Christmas.

### CODE REVIEW

**Comment by Mindengine76** [comment]
Hi Ed, sorry to say I am submitting an incomplete assignment. I hit a wall and couldn't get over it without using other people's code. I thought it was a bit disingenuous, so instead I am submitting 'faulty'  code. I am more than happy to resubmit this over Christmas, if needed.

File: oxo.py [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
+    winning_length = self.winning_length
+
+    def __init__(self, grid_width, grid_height, winning_length):
+        self.grid_width = 3
+        self.grid_height = 3
+        self.winning_length = 3
+
+        self.board = [[0 for x in xrange(0, self.grid_width)]for y in xrange(0, self.grid_height)]
+
+        self.show()
```
Comment (edpowley) [inline oxo.py:22]: Why are you calling `self.show()` here?

File: oxo.py [hunk 2]
```diff
@@ -1,35 +1,63 @@
...
+allow these squares to be appended with the characters 'x' and 'o', via the raw_input function. But this approach did 
+not match the assignment brief, so I changed to the version below. I am more than happy to re-submit this assignment 
+after Christmas, as i'm not particularly happy with what I have produced. I think this assignment clearly shows there 
+are many areas where my knowledge of programming is completely absent. This I intend to rectify."""
+
 class OxoBoard:
-    def __init__(self):
-        """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+    grid_width = self.grith_width
```
Comment (edpowley) [inline oxo.py:11]: `grith_width`?

File: oxo.py [hunk 3]
```diff
@@ -1,35 +1,63 @@
...
+    def set_square(self, x, y):
+
+       if self.board[x][y] == 0:
+          self.board[x][y] = mark
+          return True
+       else:
+           return False
+
+
+    def set_square(self, x, y, mark):
```
Comment (edpowley) [inline oxo.py:38]: Something seems to have gone a bit wrong here -- `is_board_full` has been renamed to `set_square`?

File: oxo.py [hunk 4]
```diff
@@ -1,35 +1,63 @@
...
-        raise NotImplementedError("TODO: implement set_square")
 
-    def is_board_full(self):
+    def set_square(self, x, y):
+
+       if self.board[x][y] == 0:
+          self.board[x][y] = mark
+          return True
+       else:
+           return False
```
Comment (edpowley) [inline oxo.py:35]: This looks fine for `set_square`

File: oxo.py [hunk 5]
```diff
@@ -1,35 +1,63 @@
...
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+
+        for x in xrange(0, self.grid_width):
+            for y in xrange(0, self.grid_height):
+                if self.board[x][y ] == 0:
+                    board_state = False
+                    break
+        return board_state
```
Comment (edpowley) [inline oxo.py:47]: Looks OK, though `board_state` isn't the best variable name here

**Comment by edpowley** [comment]
A reasonable start, though obviously missing some functionality. Definitely worth revisiting this exercise as your programming ability and confidence improves.

**Pull request closed without merging**